### PR TITLE
[Intel MKL] Supporting exponential_avg_factor attr for _MklFusedBatch…

### PR DESCRIPTION
**Pull request by nhasabni**
…Norm ops

This PR supports exponential_avg_factor in MklBatchNorm ops to adjust the output mean/var.

New formula is

```mean = (1 - exponential_avg_factor) * old_mean + exponential_avg_factor * batch_mean```
```var = (1 - exponential_avg_factor) * old_var + exponential_avg_factor * batch_var * adjust```

### CODE REVIEW

**Comment by nhasabni** [comment]
Hi @penpornk, this is 2nd PR. I've added the description of the change. Pls let me know if that is not enough. Thanks in advance.

File: tensorflow/core/kernels/mkl_fused_batch_norm_op_test.cc [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  this->VerifyFusedBatchNorm(exponential_avg_factor, is_training);
+}
+
+TYPED_TEST_P(FusedBatchNormOpTest, InferenceIgnoreAvgFactor) {
+  const float exponential_avg_factor = 0.5;
+  const bool is_training = false;
+  this->VerifyFusedBatchNorm(exponential_avg_factor, is_training);
+}
+
+REGISTER_TYPED_TEST_CASE_P(FusedBatchNormOpTest, Training, TrainingRunningMean,
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_fused_batch_norm_op_test.cc]: `REGISTER_TYPED_TEST_CASE_P` has be [deprecated](https://github.com/google/googletest/blob/e588eb1ff9ff6598666279b737b27f983156ad85/googletest/include/gtest/gtest-typed-test.h#L302-L308). Please use [REGISTER_TYPED_TEST_SUITE_P]( https://github.com/google/googletest/blob/e588eb1ff9ff6598666279b737b27f983156ad85/googletest/include/gtest/gtest-typed-test.h#L147) instead.

File: tensorflow/core/kernels/mkl_fused_batch_norm_op_test.cc [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                           Inference, InferenceIgnoreAvgFactor);
+
+using FusedBatchNormDataTypes = ::testing::Types<float>;
+INSTANTIATE_TYPED_TEST_CASE_P(Test, FusedBatchNormOpTest,
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_fused_batch_norm_op_test.cc]: `INSTANTIATE_TYPED_TEST_CASE_P` has been [deprecated](https://github.com/google/googletest/blob/e588eb1ff9ff6598666279b737b27f983156ad85/googletest/include/gtest/gtest-typed-test.h#L327-L333). Please use [INSTANTIATE_TYPED_TEST_SUITE_P](https://github.com/google/googletest/blob/e588eb1ff9ff6598666279b737b27f983156ad85/googletest/include/gtest/gtest-typed-test.h#L310) instead.

File: tensorflow/core/kernels/mkl_fused_batch_norm_op_test.cc [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+                              FusedBatchNormDataTypes);
+
+}  // namespace tensorflow
+#endif  // INTEL_MKL
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_fused_batch_norm_op_test.cc]: Nit: Please add a newline before this line.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! I have some comments.

**Review by penpornk — APPROVED** [review]
I'm worried about the branch cut at noon so I'm going to approve it now and make changes internally.

**Comment by nhasabni** [comment]
@penpornk Sorry I missed your review comments. If some changes still need to be done, I will address them after branch cut. Thanks.

**Comment by penpornk** [comment]
@nhasabni I added a TODO for the refactoring. All the minor fixes are done. Thank you! :)

**Comment by penpornk** [comment]
@nhasabni (And no worries!)

**Pull request merged**